packages/docs/docs/docs/model/03-encode.md

```diff
@@ -1 +1,152 @@
 # Encode
+
+As anticipated in the previous chapter, Mondrian can also automatically encode
+values of any given type. Every Mondrian type has a method `encode` to do
+that, let's look into it:
+
+```ts showLineNumbers
+const model = types.string()
+const encodingResult = model.encode("foo")
+// -> ok("foo")
+```
+
+The result of the encoding process has type `Result`, a special type
+that is returned by functions that may fail with some kind of error.
+
+In the case of encoding, the value gets validated before being encoded. Since
+the process of validation may fail for invalid values, encoding can fail as well
+if you try to encode a value that is not valid.
+
+Let's look at an example:
+
+```ts showLineNumbers
+type NonNegativeNumber = types.Infer<typeof nonNegativeNumber> // -> inferred as number
+const nonNegativeNumber = types.number({ minimum: 0 })
+
+nonNegativeNumber.encode(10) // -> ok(10)
+nonNegativeNumber.encode(-1) // -> error([ assertion: "expected a number >= 0", got: -1, path: "$" ])
+```
+
+Since additional assertions described in the type options do not change the
+inferred type, it is not possible to statically check that values of the
+inferred type respect those invariants: it is the encoder's job to make sure an
+invalid value never gets encoded.
+
+In the previous example, `NonNegativeNumber` is inferred as a simple
+Typescript `number`, so the `encode` method has to check that it is a
+non-negative number. When given a negative number the encoding process fails
+with an error describing what went wrong.
+
+## A small introduction on the `Result` type
```

Comment:
Not sure if we can leave `Result` here or if it deserves it's own chapter.

Reply:
I could also move it to its own chapter and link to it, tell me what you prefer

Reply:
Leave it there for now 👍